### PR TITLE
k8s: Wait for k8s replication controller to show up

### PR DESCRIPTION
**Pull request by c3d**
This avoids intermittent failures in the CI

Fixes: #4653

Signed-off-by: Christophe de Dinechin <dinechin@redhat.com>
Suggested-by: Greg Kurz <groug@kaod.org>

### CODE REVIEW

**Comment by c3d** [comment]
@gkurz This is essentially  your fix with a different way to set the variable

**Comment by c3d** [comment]
@gkurz @wainersm I had added this to my local CI in a loop running over the week-end. but the laptop panicked (twice, actually), so I don't know for sure how good it is.

File: integration/kubernetes/k8s-replication.bats [hunk 1]
```diff
@@ -26,8 +26,8 @@ setup() {
 	kubectl create -f "${pod_config_dir}/test-replication-controller.yaml"
 
 	# Check replication controller
-	kubectl describe replicationcontrollers/"$replication_name" | \
-		grep "replication-controller"
+        kc_rc_chk="kubectl describe replicationcontrollers/$replication_name | grep replication-controller"
+        waitForProcess "$wait_time" "$sleep_time" "$kc_rc_chk"
```
Comment (gkurz) [inline integration/kubernetes/k8s-replication.bats:30]: Rest of the script is indented with tabs. Appart from that, LGTM.

File: integration/kubernetes/k8s-replication.bats [hunk 2]
```diff
@@ -26,8 +26,8 @@ setup() {
 	kubectl create -f "${pod_config_dir}/test-replication-controller.yaml"
 
 	# Check replication controller
-	kubectl describe replicationcontrollers/"$replication_name" | \
-		grep "replication-controller"
+        kc_rc_chk="kubectl describe replicationcontrollers/$replication_name | grep replication-controller"
```
Comment (fidencio) [inline integration/kubernetes/k8s-replication.bats:29]: What does `kc_rc_chk` stands for?
I'd go for a more descriptive variable name here, @c3d.

Apart from this, LGTM.

**Review by fidencio — CHANGES_REQUESTED** [review]

File: integration/kubernetes/k8s-replication.bats (hunk 2 above)
Comment (wainersm) [reply integration/kubernetes/k8s-replication.bats:29]: or just call that `cmd` as in many other scripts. Ah, and make it a `local` variable :)

Comment (c3d) [reply integration/kubernetes/k8s-replication.bats]: Done. `cmd` is actually used a few lines below.

**Comment by c3d** [comment]
@fidencio Changed. To answer your question, `kc_rc_chk` was short `kubectl replication-controller check`. But since `cmd` is used in the same function, let's stick with that.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @c3d!

**Review by wainersm — APPROVED** [review]
Thanks @c3d !

**Comment by wainersm** [comment]
The job jenkins-ci-centos-8-stream-x86_64-crio_k8s  failed with same error as I've seen from time to time on OpenShift CI. It seems that use Stream 8 for CI isn't going to work out as this is a moving target (so subject to intermittent fails). @c3d did you create a issue about using almaLinux/Rocky instead, right?

**Comment by c3d** [comment]
@wainersm Yes, there is some ongoing work with AlmaLinux, but it's not finished yet (I spent my time today working around a libguestfs bug that impacts Ubuntu 20.04)

**Review by gkurz — APPROVED** [review]
LGTM. Thanks @c3d !

**Pull request merged**